beacon-chain/blockchain/process_block.go

```diff
@@ -601,12 +601,10 @@ func getBlockPayloadHash(blk block.BeaconBlock) ([32]byte, error) {
 
 // This saves post state info to DB or cache. This also saves post state info to fork choice store.
 // Post state info consists of processed block and state. Do not call this method unless the block and state are verified.
-func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b block.SignedBeaconBlock, st state.BeaconState, initSync bool) error {
+func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b block.SignedBeaconBlock, st state.BeaconState) error {
```

Comment:
As of now this function  only saves the block and states and for this we are copying the state and moving it around.  Does it make sense to just do these two things in `onBlock` and remove this function?

Reply:
Probably but let's do that when we clean up `onBlock` 🙏🏼